src/menu/menu.js

```diff
@@ -1,41 +1,81 @@
-// export function getMenuTemplate() {
-//   return ` <h1>Menu</h1>`;
-// }
-
-export function getRestaurantMenu() {
-  fetch(
-    `https://private-anon-7231255228-pizzaapp.apiary-mock.com/restaurants/restaurantId/menu?category=Pizza&orderBy=rank`,
-    {
-      method: "GET",
-    }
-  )
-    .then((response) => response.json())
-    .then((result) => {
-      const menuBody = document.getElementById("main");
-      result.forEach((item) => {
-        const row = document.createElement("tr");
-        const name = document.createElement("td");
-        const topping = document.createElement("td");
-        const price = document.createElement("td");
-
-        name.innerText = item.name;
-        topping.innerText = (item.topping && item.topping.join(", ")) || " ";
-        price.innerText = ` $ ${item.price}`;
-
-        row.appendChild(name);
-        row.appendChild(topping);
-        row.appendChild(price);
-
-        menuBody.appendChild(row);
+import { fetchData } from "../helpers.js";
+
+export function getMenuTemplate() {
+  return ` <h1>Menu</h1>`;
+}
+
+export function getRestaurantMenu(id) {
+  fetchData(
+    `https://private-anon-7231255228-pizzaapp.apiary-mock.com/restaurants/${id}/menu?category=Pizza&orderBy=rank`
+  ).then((result) => {
+    const menuBody = document.getElementById("main");
+
+    let totalPrice = 0;
+    let totalQuantity = 0;
+
+    result.forEach((item) => {
+      const row = document.createElement("tr");
+      const name = document.createElement("td");
+      const topping = document.createElement("td");
+      const price = document.createElement("td");
+      const addToCartBtn = document.createElement("button");
+
+      name.innerText = item.name;
+
+      topping.innerText = item.topping?.join(", ") || " ";
+      price.innerText = ` $ ${item.price}`;
+      addToCartBtn.innerText = "Add to cart";
+
+      addToCartBtn.addEventListener("click", () => {
```

Comment:
The function callback for `addToCartBtn.addEventListener ` can be moved oustside the `getRestaurantMenu ` function. This means that we will need to make `totalQuantity` and `totalPrice` global variables, meaning move them outside `getRestaurantMenu` because they are used also in the **Total Row** section.